app/src/main/kotlin/com/wire/android/ui/main/navigation/MainTopBar.kt

```diff
@@ -0,0 +1,116 @@
+package com.wire.android.ui.main.navigation
+
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.DrawerValue
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.rememberDrawerState
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.wire.android.R
+import com.wire.android.ui.common.SearchBarCollapsed
+import com.wire.android.ui.common.UserProfileAvatar
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@ExperimentalAnimationApi
+@ExperimentalComposeUiApi
+@Composable
+fun MainTopBar(scope: CoroutineScope, scaffoldState: ScaffoldState, navController: NavController, hasSearchBar: Boolean = false) {
+
+    val currentNavigationScreenItem: MainNavigationScreenItem? = navController.getCurrentNavigationItem()
+    val title = stringResource(currentNavigationScreenItem?.title ?: R.string.app_name)
+
+    ConstraintLayout(Modifier.background(MaterialTheme.colors.background)) {
+        val (topBar, searchBar) = createRefs()
+        TopAppBar(
+            modifier = Modifier.fillMaxWidth().constrainAs(topBar) {
+                top.linkTo(parent.top)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            },
+            elevation = 0.dp,
+            backgroundColor = MaterialTheme.colors.background,
+            contentColor = MaterialTheme.colors.onBackground,
+            content = {
+                ConstraintLayout(Modifier.fillMaxSize()) {
```

Comment:
do you really need this `ConstraintLayout` here? As far as TollBar content has a RowScope, you can just get rid of the additional ConstraintLayout and use 
`modifier = Modifier.weight(1f),
 textAlign = TextAlign.Center,` 
 for the title Text field